src/dbnode/server/server.go

```diff
@@ -473,29 +474,21 @@ func Run(runOpts RunOptions) {
 		SetTagEncoderPool(tagEncoderPool).
 		SetTagDecoderPool(tagDecoderPool)
 
-	db, err := cluster.NewDatabase(hostID, envCfg.TopologyInitializer, opts)
-	if err != nil {
-		logger.Fatalf("could not construct database: %v", err)
-	}
-	if err := db.Open(); err != nil {
-		logger.Fatalf("could not open database: %v", err)
-	}
-
-	// Set bootstrap options - We pass the db as the TopologyMapProvider
-	// becaues the clustered database is the component that is responsible
-	// for triggering new bootstraps based on topology changes. As a result
-	// of that, we want to make sure that when the bootstrapping process
-	// needs to capture a topology snapshot to make decisions, it gets it from
-	// the clustered database to ensure that it gets a view that is *at least*
-	// as new as the topology that triggered the bootstrap, if not newer. See
-	// GitHub issue #1013 for more details.
-	bs, err := cfg.Bootstrap.New(opts, db, origin, m3dbClient)
+	// Set bootstrap options - We need to create a topology map provider from the
+	// same topology that will be passed to the cluster so that when we make
+	// bootstrapping decisions they are in sync with the clustered database
+	// which is triggering the actual bootstraps. This way, when the clustered
+	// database receives a topology update and decides to kick off a bootstrap,
+	// the bootstrap process will receive a topology map that is at least as
+	// recent as the one that triggered the bootstrap, if not newer.
+	// See GitHub issue #1013 for more details.
+	topoMapProvider := &topoMapProvider{t: topo}
```

Comment:
nit: Create `newTopoMapProvider(...)` in case we need to validate the args, etc later?